cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -100,19 +108,91 @@ pub enum Error {
     Network(#[from] network::Error),
     #[error(transparent)]
     Wasm(#[from] wasm::Error),
+    #[error("cannot access config dir for alias file")]
+    CannotAccessConfigDir,
+    #[error(
+        "alias must be 1-30 chars long, and have only letters, numbers, underscores and dashes"
+    )]
+    InvalidAliasFormat { alias: String },
+    #[error(transparent)]
+    JSONSerialization(#[from] serde_json::Error),
+    #[error(transparent)]
+    IO(#[from] std::io::Error),
 }
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
+        self.validate_alias()?;
+
         let res = self.run_against_rpc_server(None, None).await?.to_envelope();
         match res {
             TxnEnvelopeResult::TxnEnvelope(tx) => println!("{}", tx.to_xdr_base64(Limits::none())?),
             TxnEnvelopeResult::Res(contract) => {
+                self.save_contract_id(&contract)?;
                 println!("{contract}");
             }
         }
         Ok(())
     }
+
+    fn validate_alias(&self) -> Result<(), Error> {
+        let alias = self.alias();
+
+        if alias.is_empty() {
+            return Ok(());
+        }
+
+        let regex = Regex::new(r"^[a-zA-Z0-9_-]{1,30}$").unwrap();
+
+        if regex.is_match(&alias) {
+            Ok(())
+        } else {
+            Err(Error::InvalidAliasFormat { alias })
+        }
+    }
+
+    fn alias(&self) -> String {
+        self.alias.as_ref().map(Clone::clone).unwrap_or_default()
```

Comment:
This is the same thing:
```suggestion
        self.alias.clone().unwrap_or_default()
```

Reply:
This is more readable imo. Thanks!